internal/common/environment.go

```diff
@@ -23,3 +28,16 @@ func EnvOrDefault(envName string, defVal string) string {
 func GetOperatorVersion() string {
 	return EnvOrDefault(OperatorVersionKey, defaultOperatorVersion)
 }
+
+func IsSingleNodeMode(topologyMode osconfv1.TopologyMode) bool {
+	return topologyMode == osconfv1.SingleReplicaTopologyMode
+}
+
+func GetInfrastructureTopology(c client.Client) (osconfv1.TopologyMode, error) {
+	infraConfig := &osconfv1.Infrastructure{}
```

Comment:
Openshift API needs to be added to scheme, otherwise the GET call will fail. You can add it in `internal.common.scheme` package.

Reply:
oh I did not know! Thank you!